openwisp_monitoring/device/apps.py

```diff
@@ -10,4 +14,33 @@ class DeviceMonitoringConfig(AppConfig):
     verbose_name = _('Device Monitoring')
 
     def ready(self):
+        from openwisp_controller.config.models import Device
+        from .models import DeviceData, DeviceMonitoring
         manage_short_retention_policy()
+        health_status_changed.connect(
+            self.set_cache_key,
+            sender=DeviceMonitoring
+        )
+        if getattr(settings, 'OPENWISP_MONITORING_DEVICE_RECOVERY_DETECTION', True):
+            checksum_requested.connect(
+                self.check_cache_key,
+                sender=Device,
+                dispatch_uid='checksum_requested'
+            )
+            device_metrics_received.connect(
+                self.check_cache_key,
+                sender=DeviceData,
+                dispatch_uid='received_device_metrics'
+            )
+
+    @classmethod
+    def set_cache_key(cls, instance, status, **kwargs):
```

Comment:
I'd use `manage_device_recovery_cache_key` and add a docstring to this method explains why it sets or remove this cache key